library/common/network/configurator.cc

```diff
@@ -82,6 +82,11 @@ Configurator::NetworkState Configurator::network_state_{1, ENVOY_NET_GENERIC, Ma
 
 envoy_netconf_t Configurator::setPreferredNetwork(envoy_network_t network) {
   Thread::LockGuard lock{network_state_.mutex_};
+  if (network_state_.network_ == network) {
```

Comment:
I think that this check will break DNS a network change update for when we move from no-internet connection to wi-fi connection and NWPathMonitor is _disabled_. This is because of this logic that does not differentiate between the lack of connection and wi-fi connection https://github.com/envoyproxy/envoy-mobile/blob/main/library/objective-c/EnvoyNetworkMonitor.m#L125-L132

Reply:
We can update `NetworkPathMonitor` so that it properly differentiates between no-connection and wi-fi connection but in general addition of this check does not seem to be related to the changes in this PR and it does increase a risk of the changes in this PR- especially as this functionality is not behind a FF.

Reply:
I don't quite understand this change in general? Can you add more comments? We can also discuss in our sync.

Reply:
Added further comments, but also disabled this check for now.